src/to_vec.jl

```diff
@@ -4,42 +4,41 @@
 Transform `x` into a `Vector`, and return the vector, and a closure which inverts the
 transformation.
 """
-function to_vec(x::Number)
+function to_vec(x::T) where {T<:Number}
     function Number_from_vec(x_vec)
-        return first(x_vec)
+        return T(first(x_vec))
     end
     return [x], Number_from_vec
 end
 
-# Vectors
+# (Abstract)Vectors
 to_vec(x::Vector{<:Number}) = (x, identity)
-function to_vec(x::Vector)
+
+function to_vec(x::T) where {T<:AbstractVector}
     x_vecs_and_backs = map(to_vec, x)
     x_vecs, backs = first.(x_vecs_and_backs), last.(x_vecs_and_backs)
     function Vector_from_vec(x_vec)
         sz = cumsum(map(length, x_vecs))
-        return [backs[n](x_vec[sz[n] - length(x_vecs[n]) + 1:sz[n]]) for n in eachindex(x)]
+        x_Vec = [backs[n](x_vec[sz[n] - length(x_vecs[n]) + 1:sz[n]]) for n in eachindex(x)]
+        return T(x_Vec)
```

Comment:
Here i think it is more important to use `oftype` because it calls out to `convert` which is more appropriate than just assuming a constructor would exist taking this arg.
```suggestion
        return oftype(x, x_Vec)
```

Because `oftype` using `convert`